File: models/Blog.go
package models

import "go.mongodb.org/mongo-driver/bson/primitive"

// Blog model
type Blog struct {
	ID       primitive.ObjectID `bson:"_id,omitempty"`
	Title    string             `bson:"title"`
	Content  string             `bson:"content"`
	Author   string             `bson:"author"`
	Comments []Comment          `bson:"comments"`
}

// Comment model
type Comment struct {
	Username string `bson:"username"`
	Content  string `bson:"content"`
}

File: models/Blog.js
const mongoose = require('mongoose');

const commentSchema = new mongoose.Schema({
  username: { type: String, required: true },
  text: { type: String, required: true },
  // createdAt: { type: Date, default: Date.now },
});

const blogSchema = new mongoose.Schema({
  title: { type: String, required: true },
  content: { type: String, required: true },
  author: { type: String, required: true },
  createdAt: { type: Date, default: Date.now },
  comments: [commentSchema],
  img: { type: String, required: true }
});

const Blog = mongoose.model('Blog', blogSchema, 'Blog');

module.exports = Blog;
